app/static/src/app/components/options/RunOptions.vue

```diff
@@ -5,9 +5,12 @@
         <label class="col-form-label">End time</label>
       </div>
       <div class="col-6">
-        <numeric-input :value="endTime"
+        <numeric-input
+                       v-if="endTimeData === 0"
+                       :value="endTime"
                        :allow-negative="false"
                        @update="updateEndTime"></numeric-input>
+        <label v-if="endTimeData > 0" class="col-form-label">{{ endTimeData }} (from data)</label>
```

Comment:
We shouldn't ever have negative end times so `v-else` should be safe
```suggestion
        <label v-else class="col-form-label">{{ endTimeData }} (from data)</label>
```

Reply:
thanks, I did not know about this!